torch/csrc/jit/runtime/static/impl.h

```diff
@@ -454,5 +456,57 @@ class TORCH_API ProcessedNode {
   std::vector<IValue> outputs_;
 };
 
+//  Map each value to all values that are alive at the same time.
+using LivenessMap = FastMap<const Value*, std::set<const Value*>>;
+
+typedef struct LiveRange {
+  size_t begin;
+  size_t end;
+
+} LiveRange;
+
+inline std::ostream& operator<<(std::ostream& str, LiveRange lvr) {
+  return str << "[" << lvr.begin << ", " << lvr.end << "]";
+}
+
+struct live_range_start_cmp {
+  bool operator()(LiveRange const& lvr1, LiveRange const& lvr2) const {
+    return lvr1.begin == lvr2.begin ? lvr1.end < lvr2.end
+                                    : lvr1.begin < lvr2.begin;
+  }
+};
+
+struct live_range_end_cmp {
+  bool operator()(LiveRange const& lvr1, LiveRange const& lvr2) const {
+    return lvr1.end == lvr2.end ? lvr1.begin < lvr2.begin : lvr1.end < lvr2.end;
+  }
+};
+
+struct live_range_hash {
+  size_t operator()(LiveRange const& range) const {
+    return std::hash<size_t>()(range.begin) ^
+        (std::hash<size_t>()(range.end) << 1);
+  }
+};
+
+inline bool operator==(const LiveRange& lhs, const LiveRange& rhs) {
+  return lhs.begin == rhs.begin && lhs.end == rhs.end;
+}
+
+inline bool operator!=(const LiveRange& lhs, const LiveRange& rhs) {
+  return !(lhs == rhs);
+}
+
+using LiveRangesMap = std::unordered_map<const Value*, LiveRange>;
+
+TORCH_API std::unordered_set<const Value*> GetAlwaysAliveValues(
```

Comment:
Return type doesn't match the the implementation.

Reply:
ah sorry `FastSet <=> unordered_set` got away from me.